### PR TITLE
Add a commit-msg checker

### PR DESCRIPTION
**Pull request by Rapiz1**
Add a commit-msg checker

Resolves https://github.com/apache/incubator-apisix-dashboard/issues/140

@juzhiyuan Can you test this on Mac?

### CODE REVIEW

File: scripts/verifyCommit.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// Invoked on the commit-msg git hook by yorkie.
+
+const chalk = require('chalk');
+const msgPath = process.env.HUSKY_GIT_PARAMS;
+const msg = require('fs').readFileSync(msgPath, 'utf-8').trim();
+
+const commitRE = /^(revert: )?(feat|fix|docs|dx|style|refactor|perf|test|workflow|build|ci|chore|types|wip|release)(\(.+\))?: .{1,50}/;
+
+if (!commitRE.test(msg)) {
+  console.log();
```
Comment (juzhiyuan) [inline scripts/verifyCommit.js:33]: empty log?

File: scripts/verifyCommit.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  console.error(
+    `  ${chalk.bgRed.white(' ERROR ')} ${chalk.red(`invalid commit message format.`)}\n\n` +
+      chalk.red(
+        `  Proper commit message format is required for automated changelog generation. Examples:\n\n`,
+      ) +
+      `    ${chalk.green(`feat(compiler): add 'comments' option`)}\n` +
+      `    ${chalk.green(`fix(v-model): handle events on blur (close #28)`)}\n\n` +
+      chalk.red(`  See .github/commit-convention.md for more details.\n`),
```
Comment (juzhiyuan) [inline scripts/verifyCommit.js:41]: what's `.github/X`?

**Comment by juzhiyuan** [comment]
> Add a commit-msg checker
> 
> Resolves #140
> 
> @juzhiyuan Can you test this on Mac?

Not yet. I switched to Windows those days..

File: scripts/verifyCommit.js (hunk 2 above)
Comment (Rapiz1) [reply scripts/verifyCommit.js:41]: That is from vue-next. Should I remove this line or add that file?

File: scripts/verifyCommit.js (hunk 1 above)
Comment (Rapiz1) [reply scripts/verifyCommit.js:33]: This is for a blank line in the error. See 
image

**Comment by Rapiz1** [comment]
> Not yet. I switched to Windows those days..

Not a big deal. This will probably works well on Mac according to [here](https://github.com/typicode/husky/issues/113).

File: scripts/verifyCommit.js (hunk 2 above)
Comment (juzhiyuan) [reply scripts/verifyCommit.js:41]: emmm not sure

Comment (Rapiz1) [reply scripts/verifyCommit.js:41]: I'm think it's a good idea to follow the practice of vue-next and so on opensource projects. It's kind of confusing if we have a commit-checker but lacks of the introduction to commit convention.

Comment (Rapiz1) [reply scripts/verifyCommit.js:41]: So I will add that if you don't have any doubts.

Comment (juzhiyuan) [reply scripts/verifyCommit.js:41]: Just go ahead!

File: .github/commit-convention.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Appears under "Features" header, `compiler` subheader:
+
+```
+feat(compiler): add 'comments' option
+```
+
+Appears under "Bug Fixes" header, `v-model` subheader, with a link to issue #28:
+
+```
+fix(v-model): handle events on blur
```
Comment (juzhiyuan) [inline .github/commit-convention.md:24]: v-model?

Comment (Rapiz1) [reply .github/commit-convention.md:24]: I think it's just an example and does not mean any restraint to commit message.

**Pull request merged**